pkg/volume/flexvolume/probe.go

```diff
@@ -127,48 +173,59 @@ func (prober *flexVolumeProber) handleWatchEvent(event fsnotify.Event) error {
 	if err != nil {
 		return err
 	}
-
+	parentPathAbs := filepath.Dir(eventPathAbs)
 	pluginDirAbs, err := filepath.Abs(prober.pluginDir)
 	if err != nil {
 		return err
 	}
 
-	// If the Flexvolume plugin directory is removed, need to recreate it
-	// in order to keep it under watch.
-	if eventOpIs(event, fsnotify.Remove) && eventPathAbs == pluginDirAbs {
-		if err := prober.createPluginDir(); err != nil {
-			return err
-		}
-		if err := prober.addWatchRecursive(pluginDirAbs); err != nil {
-			return err
+	// event about pluginDirAbs
+	if eventPathAbs == pluginDirAbs {
+		// If the Flexvolume plugin directory is removed, need to recreate it
+		// in order to keep it under watch.
+		if eventOpIs(event, fsnotify.Remove) {
+			if err := prober.createPluginDir(); err != nil {
+				return err
+			}
+			if err := prober.addWatchRecursive(pluginDirAbs); err != nil {
+				return err
+			}
 		}
-	} else if eventOpIs(event, fsnotify.Create) {
-		if err := prober.addWatchRecursive(eventPathAbs); err != nil {
-			return err
+	}
+
+	// event about subdir of pluginDirAbs
+	if parentPathAbs == pluginDirAbs {
```

Comment:
If anything inside a driver directory changes, even if it's many levels of subdirectories in, the driver should re-initialize. With your logic, newly added subdirectories inside a driver directory won't be watched. I believe the existing logic already works, please correct me if I'm wrong.

Reply:
Yes, you are right.